### DISCUSSION TITLE
Trigger python script

### DISCUSSION THREAD

**Discussion by Marzocchi666** [TwiN/gatus#338] [Q&A] [answered]
Hello,

we would like to add gatus into our ecosystem, and we are wondering if there is a possibility to trigger a script from gatus, which outputs some data, and based on the data resulted from the script to add conditions in gatus. 

Thank you,
Marzocchi

**Comment by TwiN** [comment]
Similar, but for alerting: https://github.com/TwiN/gatus/issues/118

There's still some work needed before we can even start working on this, but we're not _too far_. Still, I prioritize issues based on what I want to work on (sometimes I just feel like tackling a specific problem), and based on the number of 👍 on existing isssues. As the issue I've mentioned doesn't have that many 👍, I don't think I'll be working on this very soon.

---

I could have sworn there was an issue for providing an API to manually insert an endpoint's status, but it appears there isn't. That would be the eventual solution to your problem - you could just have a python script run on a cronjob that would do its thing, and then call Gatus' endpoint to update the endpoint.

**Reply by Marzocchi666** [reply]
In my use case, i have some data which would then feed gatus status page, and im trying to find an optimal and efficient solution for this. So, gatus should somehow send some sort of signal and based on that the script should be triggered and refresh the page with new data. Or i could send the python script data to some monitoring api, then pick up the data into gatus from there. 
I appreciate your reply to my post!! 
Have a great day,
Marzocchi!

**Comment by TwiN** [comment, marked as answer]
This is currently "kind of" possible by using external endpoints to push status instead of having Gatus be the one to run the script.

See By default, the configuration file is expected to be at config/config.yaml.

You can specify a custom path by setting the GATUS_CONFIG_PATH environment variable.

If GATUS_CONFIG_PATH points to a directory, all *.yaml and *.yml files inside said directory and its subdirectories are merged like so:

    All maps/objects are deep merged (i.e. you could define alerting.slack in one file and alerting.pagerduty in another file)
    All slices/arrays are appended (i.e. you can define endpoints in multiple files and each endpoint will be added to the final list of endpoints)
    Parameters with a primitive value (e.g. debug, metrics, alerting.slack.webhook-url, etc.) may only be defined once to forcefully avoid any ambiguity
        To clarify, this also means that you could not define alerting.slack.webhook-url in two files with different values. All files are merged into one before they are processed. This is by design.
